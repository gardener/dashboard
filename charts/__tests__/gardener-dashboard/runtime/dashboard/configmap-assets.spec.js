//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm, helper } = fixtures
const { encodeBase64 } = helper

const renderTemplates = helm.renderDashboardRuntimeTemplates

describe('gardener-dashboard', function () {
  describe('configmap-assets', function () {
    let templates

    beforeEach(() => {
      templates = [
        'configmap-assets'
      ]
    })

    it('should render the template', async function () {
      const values = {
        global: {
          dashboard: {
            frontendConfig: {
              assets: {
                'bar.png': encodeBase64('foo\n')
              }
            }
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [assets] = documents
      expect(assets).toMatchSnapshot()
    })
  })
})
