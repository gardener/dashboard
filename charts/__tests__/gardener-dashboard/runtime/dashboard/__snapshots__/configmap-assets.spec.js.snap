// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard configmap-assets should render the template 1`] = `
Object {
  "apiVersion": "v1",
  "binaryData": Object {
    "bar.png": "Zm9vCg==
",
  },
  "kind": "ConfigMap",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "dashboard-assets",
    "namespace": "garden",
  },
}
`;
